tests/test_github.ml

```diff
@@ -1,59 +1,65 @@
-    | Error (`Msg msg) -> Alcotest.fail msg
+open Bos_setup
+
+let curl = Cmd.(v "curl" % "-L" % "-s" % "-S" % "-K" % "-")
+
+let test_run_with_auth =
+  let make_test name ~auth ~curl ~expected =
+    let test_name = "run_with_auth " ^ name in
+    let test_fun () =
+      match
+        Dune_release.Github.run_with_auth ~dry_run:false ~default:"NONE" auth
+          curl
+      with
+      | Ok r -> (
+          match Dune_release.Json.from_string r with
+          | Ok json -> (
+              match Dune_release.Github_v3_api.error_message json with
+              | Some s -> Alcotest.(check string) name expected s
+              | None -> Alcotest.(check string) name expected "VALID JSON" )
+          | Error (`Msg e) -> Alcotest.(check string) name expected e )
+      | Error (`Msg e) -> Alcotest.(check string) name expected e
+    in
+    (test_name, `Quick, test_fun)
   in
-  check
-    ( "{"
-    ^ {|"url":"https://api.github.com/repos/NathanReb/dune-release-testing/releases/assets/12789323",|}
-    ^ {|"id":12789323,|} ^ {|"node_id":"MDEyOlJlbGVhc2VBc3NldDEyNzg5MzIz",|}
-    ^ {|"name":"dummy-v0.0.0.tbz",|} ^ {|"label":"",|}
-    ^ {|"uploader":{"login":"NathanReb",|} ^ {|"id":7419360,|}
-    ^ {|"node_id":"MDQ6VXNlcjc0MTkzNjA=",|}
-    ^ {|"avatar_url":"https://avatars2.githubusercontent.com/u/7419360?v=4",|}
-    ^ {|"gravatar_id":"",|}
-    ^ {|"url":"https://api.github.com/users/NathanReb",|}
-    ^ {|"html_url":"https://github.com/NathanReb",|}
-    ^ {|"followers_url":"https://api.github.com/users/NathanReb/followers",|}
-    ^ {|"following_url":"https://api.github.com/users/NathanReb/following{/other_user}",|}
-    ^ {|"gists_url":"https://api.github.com/users/NathanReb/gists{/gist_id}",|}
-    ^ {|"starred_url":"https://api.github.com/users/NathanReb/starred{/owner}{/repo}",|}
-    ^ {|"subscriptions_url":"https://api.github.com/users/NathanReb/subscriptions",|}
-    ^ {|"organizations_url":"https://api.github.com/users/NathanReb/orgs",|}
-    ^ {|"repos_url":"https://api.github.com/users/NathanReb/repos",|}
-    ^ {|"events_url":"https://api.github.com/users/NathanReb/events{/privacy}",|}
-    ^ {|"received_events_url":"https://api.github.com/users/NathanReb/received_events",|}
-    ^ {|"type":"User",|} ^ {|"site_admin":false},|}
-    ^ {|"content_type":"application/x-tar",|} ^ {|"state":"uploaded",|}
-    ^ {|"size":811,|} ^ {|"download_count":0,|}
-    ^ {|"created_at":"2019-05-21T09:27:22Z",|}
-    ^ {|"updated_at":"2019-05-21T09:27:22Z",|}
-    ^ {|"browser_download_url":"https://github.com/NathanReb/dune-release-testing/releases/download/v0.0.0/dummy-v0.0.0.tbz"|}
-    ^ "}" )
-    "https://github.com/NathanReb/dune-release-testing/releases/download/v0.0.0/dummy-v0.0.0.tbz"
+  [
+    make_test "invalid command" ~auth:""
+      ~curl:
+        (let uri =
+           "https://api.github.com/repos/NathanReb/dune-release-testing/releases/latest"
+         in
+         Cmd.(curl % uri))
```

Comment:
I don't think that's a good idea to hit the github API in the unit tests, as you probably saw already, we have race conditions due to rate limiting.

If you want to unit test the parsing of curl's output I suggest testing it on an example string.

We can later add integration tests that try to release an actual test repo to our opam-repository fork but those should never be run in the CI, or at least not as part of `dune runtest` which is ran by opam.

Reply:
Also if we want to do this, I'd rather not do it through `Alcotest` as that wouldn't be unit tests.
I think such integration tests are out of scope of this PR.

Reply:
Agreed, I removed these tests.